include/layers/Tensor.hpp

```diff
@@ -21,11 +21,12 @@ template <typename T>
 const std::vector<uint8_t>* to_byte(const std::vector<T>& v) {
   return reinterpret_cast<const std::vector<uint8_t>*>(&v);
 }
-
 template <typename T>
 Type GetTypeEnum() {
   if constexpr (std::is_same_v<T, int>) {
     return Type::kInt;
+  } else if constexpr (std::is_same_v<T, int64_t>) {
```

Comment:
`GetTypeEnum<int64_t>()` returns `Type::kInt` (same as 32‑bit). This loses int32/int64 distinction; consider a separate enum value or explicit casts at boundaries.